streams/src/test/java/org/apache/kafka/streams/integration/RegexSourceIntegrationTest.java

```diff
@@ -227,28 +226,27 @@ public boolean conditionMet() {
         }, STREAM_TASKS_NOT_UPDATED);
     }
 
-    @SuppressWarnings("deprecation")
     @Test
     public void shouldAddStateStoreToRegexDefinedSource() throws InterruptedException {
 
         final ProcessorSupplier<String, String> processorSupplier = new MockProcessorSupplier<>();
-        final MockStateStoreSupplier stateStoreSupplier = new MockStateStoreSupplier("testStateStore", false);
+        final StoreBuilder storeBuilder = Stores.keyValueStoreBuilder(Stores.inMemoryKeyValueStore("testStateStore"), Serdes.String(), Serdes.String());
         final long thirtySecondTimeout = 30 * 1000;
 
-        final TopologyBuilder builder = new TopologyBuilder()
+        final Topology topology = new TopologyWrapper()
```

Comment:
use `TopologyWrapper` on left hand side and avoid cast below?

Reply:
Unfortunately I cannot since once the `addXX` is called the returned type is `Topology`. But I refactored the code a bit to not use chained operators:

```
        final TopologyWrapper topology = new TopologyWrapper();
        topology.addSource("ingest", Pattern.compile("topic-\\d+"));
        topology.addProcessor("my-processor", processorSupplier, "ingest");
        topology.addStateStore(storeBuilder, "my-processor");
```